awx/api/serializers.py

```diff
@@ -3293,19 +3349,23 @@ def get_defaults(self, obj):
         ask_for_vars_dict['vault_credential'] = False
         defaults_dict = {}
         for field in ask_for_vars_dict:
-            if field in ('inventory', 'credential', 'vault_credential'):
+            if field == 'inventory':
                 defaults_dict[field] = dict(
                     name=getattrd(obj, '%s.name' % field, None),
                     id=getattrd(obj, '%s.pk' % field, None))
-            elif field == 'extra_credentials':
+            elif field in ('credential', 'vault_credential', 'extra_credentials'):
```

Comment:
This is one backwards incompatible change, and I think it makes sense.  Now if you visit `/api/v2/job_templates/launch/`, the `defaults` key will _not_ include `credential`, et. al

The code *does* allow you to pass these in a backwards-compatible way, but now we only populate `defaults` with new-style `credentials` to encourage its use.

Reply:
This is definitely one of the bits the UI team may need to know about explicitly.

Reply:
This is in fact the one thing I've found in the UI that currently _does not_ work with this PR - the UI looks for the "old" key values to prefill the launch dialog, so they'll need to change that to use this new `credentials` key instead.  I hesitated to add both because I didn't want people to accidentally *submit* both on `POST /api/v2/job_templates/N/launch/` (because that's not valid: https://github.com/ansible/awx/pull/595/files/47159b083cc32ec72927ed6621ba91a914dde5db#diff-be9a06acc0c9b5d11468e143d3dae1b9R2748)

Reply:
:thumbsup: thanks for the heads up. I think this is the place that would need to change:

https://github.com/ansible/awx/blob/devel/awx/ui/client/src/job-submission/job-submission.controller.js#L165

Reply:
@jakemcdermott yep, that's about what I expected 👍